### PR TITLE
Move unsupported files notification into problems console #173

### PR DESCRIPTION
**Pull request by Nurkambay**
- Move unsupported files notification into problems console instead of logging
- Add unit tests for uncovered methods

### CODE REVIEW

File: server/pom.xml [hunk 1]
```diff
@@ -168,6 +168,12 @@
             <version>${mockito.core.version}</version>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (Nurkambay) [inline server/pom.xml:171]: Do we need to add new TPSR for it?

File: server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java [hunk 2]
```diff
@@ -16,42 +16,157 @@
...
+
+  @Mock
+  private HashSet<String> uriInProgress;
+
+  @InjectMocks
+  private ServerCommunications communications;
+
+  @BeforeEach
+  void init(TestInfo info) {
+    if (!info.getDisplayName().equals("testCancelProgressNotification()")) {
```
Comment (asatklichov) [inline server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java:62]: safer check: !"testCancelProgressNotification()".equals(info.getDisplayName())

**Comment by temanbrcom** [comment]
@asatklichov

File: server/src/main/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunications.java [hunk 3]
```diff
@@ -109,7 +115,7 @@ public void notifyThatDocumentAnalysed(String uri) {
    */
   @Override
   public void notifyThatExtensionIsUnsupported(String extension) {
-    runAsync(() -> showMessage(Error, "The given document extension is unsupported: " + extension));
+    runAsync(() -> logMessage(Error, EXTENSION_UNSUPPORTED + extension));
```
Comment (Nurkambay) [inline server/src/main/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunications.java:118]: The actual fix

**Review by temanbrcom — CHANGES_REQUESTED** [review]

File: server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java [hunk 4]
```diff
@@ -16,42 +16,157 @@
...
+   * Method {@link ServerCommunications#publishDiagnostics(Map)} should
+   * raise a diagnostic message to the client with syntax error retrieved by the COBOL
+   * LSP server for related files.
+   */
+  @Test
+  void testPublishDiagnostics() {
+    String uri = UUID.randomUUID().toString();
+
+    // Prepare diagnostic map
+    Map<String, List<Diagnostic>> map = new HashMap<>();
```
[the remark this answers is not in the document]
Comment (Nurkambay) [reply server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java]: Fixed

File: server/src/main/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunications.java [hunk 5]
```diff
@@ -43,8 +43,14 @@
...
+  static final String CANNOT_FIND_LANGUAGE_ENGINE = "Cannot find a language engine for the given language ID: ";
+  static final String SYNTAX_ANALYSIS_IN_PROGRESS = ": Syntax analysis in progress";
+  static final String NO_SYNTAX_ERRORS = "No syntax errors detected in ";
+  static final String EXTENSION_UNSUPPORTED = "The given document extension is unsupported: ";
+
+  private final Provider<LanguageClient> provider;
+  private final FileSystemService files;
 
   private final ScheduledExecutorService executor = Executors.newScheduledThreadPool(5);
   private final Set<String> uriInProgress = new HashSet<>();
```
[the remark this answers is not in the document]
Comment (Nurkambay) [reply server/src/main/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunications.java]: Fixed

**Review by temanbrcom — APPROVED** [review]
🚜

**Review by temanbrcom — APPROVED** [review]

File: server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java [hunk 6]
```diff
@@ -16,42 +16,156 @@
...
+   * LSP server for related files.
+   */
+  @Test
+  void testPublishDiagnostics() {
+    String uri = UUID.randomUUID().toString();
+
+    // Prepare diagnostic map
+    Diagnostic diagnostic = new Diagnostic(new Range(), "\r\ntest\r\n");
+    List<Diagnostic> diagnostics = List.of(diagnostic);
+    //map.put(uri, diagnostics);
```
Comment (asatklichov) [inline server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java:139]: clean-up needed

**Review by asatklichov — APPROVED** [review]

File: server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java (hunk 6 above)
Comment (Nurkambay) [reply server/src/test/java/com/broadcom/lsp/cobol/service/delegates/communications/ServerCommunicationsTest.java]: Fixed

**Pull request merged**